### ISSUE TITLE
API Lifecycle Weekly - January 6th, 2022

### ISSUE THREAD

**Issue by kinlane** [postman-open-technologies/lifecycle#118]
This is a weekly public meeting to discuss the different areas of the API lifecycle and governance, push forward discussions around how the API lifecycle applies to OpenAPI and AsyncAPI, and work on extensions, platform ops collections, tooling, and blueprints that will help the community be more successful.

**Agenda:**

- **Overview** ([Deck](https://docs.google.com/presentation/d/12j5E9Heb9OEkpHqlOIePlGmq80e0qms6G3SKbj96NPA/edit#slide=id.gf6090eea14_0_20)) - Walk through the 100K of what we are looking to accomplish with this lifecycle work.
- **APIs.How** ([Workspace](https://www.postman.com/postman/workspace/postman-open-technologies-lifecycle/overview)) - Walkthrough what the lifecycle blueprint strategy is, and how we can evolve it to better help educate the community.
- **Platform Ops** ([Specifications](https://www.postman.com/postman/workspace/postman-open-technologies-specifications/overview), [Design](https://www.postman.com/postman/workspace/postman-open-technologies-design/overview), [Mocking](https://postman.postman.co/workspace/Postman-Open-Technologies---Moc~dc355964-3e10-4f65-bfcd-d960b55bd7f4/overview), [Documentation](https://postman.postman.co/workspace/Postman-Open-Technologies---Doc~0cc45f27-9ef7-4165-a228-e4a5b0f606f9/overview), [Environments](https://postman.postman.co/workspace/Postman-Open-Technologies---Env~26804961-0893-442f-8d90-9a3fe1b10b9f/overview), [Testing](https://postman.postman.co/workspace/Postman-Open-Technologies---Tes~687d93f0-a8d3-488b-b37b-db968561fe08/overview), [Security](https://postman.postman.co/workspace/Postman-Open-Technologies---Sec~38535780-4999-4a6c-858d-4c80e40d9df2/overview), [Monitoring](https://postman.postman.co/workspace/Postman-Open-Technologies---Mon~c100a23f-4b34-4afd-8213-5d6de59cd224/overview), [Discovery](https://postman.postman.co/workspace/Postman-Open-Technologies---Dis~66ef7f43-ebf8-4892-b3c5-fdaefbf535a9/overview), [Workspaces](https://www.postman.com/postman/workspace/postman-open-technologies-workspaces), [Client](https://postman.postman.co/workspace/Postman-Open-Technologies---Cli~f9960048-1490-4028-956f-87d120860b81/overview), [Source Control](https://postman.postman.co/workspace/Postman-Open-Technologies---Sou~49eb94f6-5f85-453a-acf9-0c4001013a88/overview), [CI/CD](https://www.postman.com/postman/workspace/postman-open-technologies-ci-cd/overview), [Gateway](https://www.postman.com/postman/workspace/postman-open-technologies-gateways), [APM](https://www.postman.com/postman/workspace/postman-open-technologies-apm/overview), [Orchestration](https://postman.postman.co/workspace/Postman-Open-Technologies---Orc~1da91497-f032-44a1-9368-8b11aea95907), [Observability](https://postman.postman.co/workspace/Postman-Open-Technologies---Obs~c532af03-f621-4f84-ad5a-b5212b79eab2/overview), [Governance](https://www.postman.com/postman/workspace/postman-open-technologies-governance), [Analytics](https://postman.postman.co/workspace/Postman-Open-Technologies---Ana~43665cec-d41a-412e-913d-dc15e1f1909d/overview), [Teams](https://postman.postman.co/workspace/Postman-Open-Technologies---Tea~7cedb91f-d09e-416d-8308-0cb180d2835b/overview),[Change](https://www.postman.com/postman/workspace/postman-open-technologies-change), [Automation](https://www.postman.com/postman/workspace/postman-open-technologies-automation)) - Outline how we’ll be defining each area of the API lifecycle, then establishing platform operation collections for each area.
- **Governance** ([Workspace](https://www.postman.com/postman/workspace/postman-open-technologies-governance)) - Move forward with storytelling around [linting rules](https://www.postman.com/postman/workspace/postman-open-technologies-governance-rules/overview), and how to apply them as part of the API lifecycle in service of governance.
- **Templates** ([Workspace](https://www.postman.com/postman/workspace/published-postman-templates/overview)) - Explore more templates that we can develop, how we can turn templates, standards, and common practices into more of a production.
- **State of the API** ([Report](https://www.postman.com/state-of-api/))- Discuss how we can make the State of the API report more hands-on, with regular storytelling and discussions around the findings.

Please add any agenda items you’d like to discuss below, and we will meet weekly, and push forward projects, stories, collections, and tools to support our work.

Meeting:

- January 6th, 2022 - 8:00 AM PST 
- [Zoom](https://us02web.zoom.us/j/88645145164)

**Comment by kevinswiber** [comment]
Notes from API Lifecycle Weekly - 2022-01-06

tl;dr: It’s okay to be “hand-wavy”, as long we’re talking about it publicly.
 
APIs.how:
* Review the Lifecycle portion and how it fits into the Postman Open Technologies mission.
* Matthew: Via a question on LinkedIn, who’s authoring the book? 
* APIs.how: Kin is working with Sales on 60-second videos explaining each Blueprint.  Need to add a visual layer.
* Platform Ops and Governance offer an opportunity to lead the conversation.

Platform Ops:
* Public workspace for each area of Platform Ops.
* What are some ops collections we want to focus on?  Gateway, CI/CD, Source control are the top 3.  Security, monitoring, discovery, documentation, and orchestration are also significant.
* How do we extract ops collections that are meaningful?  Example: 42Crunch.
* Let’s be iterative in our approach to delivery.  It’s okay if a collection isn’t verified on live-running infrastructure.  Let’s just get it out there and build upon it.  Let’s put out a conversation starter.
* Something about Kin being a steward on an Open Technologies transportation vessel?

Governance:
* Not a clear message, and we need to remedy that.
* Linting Rules: 150 or so OpenAPI rules for spec linting. Not all of them are certified working.  Need to automate the validation.
* Only one (1) cough in this section.
* Only one (1) sentence requires a parental advisory sticker on the recording.
* Rules-based linting, script-based linting.  Design time, collection, pipeline, CLI, IDE.  Linting everywhere.
* Hoping to communicate “our governance stance” in the next week or 2.
* The narrative is still fuzzy.  
* A complete list of rules/blueprints/etc. lives in a Lifecycle Control workspace in Postman.
* Matthew is working on 2 rulesets: 1 for Stripe, 1 for Twilio.  Will work on more storytelling.

Templates:
* A set of templates that can be used to jumpstart the implementation of different patterns and standards.
* Rate-limiting is going in today.
* Blog post is ready that’s just waiting for art assets before being published.  Matthew has decent traction with the community with respect to providing feedback.
* Own the conversation around this.  Bring into the workspace, into the community.  Beat every drum.
* Matthew’s RFC proposal: Allow organizations to populate new API definitions by default in Postman.

State of the API
* Nope.  Gonna do it next week.  We’re out of time!
* …And then Matthew said something about stitching together GitHub something-something.  Oh no.  I’m gonna get caught spacing out at the end.

Action Items:
* Kin will check with Abhinav, Ankit, and Abhijit on who will be credited as authors for the upcoming book.
* Kin will ping Noname, Salt Security and ask about their APIs.
* Meena will keep creating service provider collections, and we can have conversations about the findings.
* Kevin will keep working to define a template for what an ops collection can be. How does it fit into the SDLC?  What’s the scope?  Fine-grained, coarse-grained? If we come up with a solution for one provider, how does that make its away across other providers?  How does that impact storytelling?
* Meena and Kevin will bring thoughts to share regarding the ops collections creation process.
* Kin + Kevin: Get the Linting API hosted on a domain.
* Kevin to figure out using Newman/collection vs CLI approach to API linting.
* Stabilize the message.  2 blog posts in January: 1) Entrance into governance space and 2) What are Spectral rules?

**Comment by kinlane** [comment]
[![API Lifecycle Meeting](https://i9.ytimg.com/vi/FaMrxZB1uZs/mq1.jpg?sqp=COSI3Y4G&rs=AOn4CLCp9nEFT6A3x5RxbtAvtMY3mKLOKg)](https://www.youtube.com/watch?v=FaMrxZB1uZs)
